src/libraries/Microsoft.Extensions.Hosting/src/Host.cs

```diff
@@ -111,8 +111,11 @@ public static IHostBuilder CreateDefaultBuilder(string[] args)
 
                 if (isWindows)
                 {
+#pragma warning disable CA1416 // Platform compat analyzer
```

Comment:
Why is this necessary? `isWindows` is defined above as:

```
bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```

Can the analyzer not recognize this pattern?

Reply:
Analyzer can track result save in local, it shouldn't warn

Reply:
It was warning on this.  Remind me; can the analyzer also recognize the old `RuntimeInformation.IsOSPlatform()` calls like this one? This branch is still intact despite the PR being close.  Could you pull the branch down and check this scenario please, @buyaa-n?

Reply:
> can the analyzer also recognize the old RuntimeInformation.IsOSPlatform()

Yes it can

> It was warning on this. 

That is strange, i added exact same test and it is not warning

> This branch is still intact despite the PR being close. Could you pull the branch down and check this scenario please, @buyaa-n?

Sure

Reply:
Filed an issue for the bug https://github.com/dotnet/runtime/issues/41535